src/backends/caffe2/autofill.cc

```diff
@@ -78,12 +78,17 @@ AutoFillNetDef::Create(
   const std::string expected_init_filename =
       std::string(kCaffe2NetDefInitFilenamePrefix) +
       std::string(kCaffe2NetDefFilename);
-  if (netdef_files.size() != 2) {
+  if (netdef_files.size() < 2) {
     return Status(
         RequestStatusCode::INTERNAL, "unable to autofill for '" + model_name +
                                          "', unable to find netdef files: '" +
                                          kCaffe2NetDefFilename + "' and '" +
                                          expected_init_filename + "'");
+  } else if (netdef_files.size() > 2) {
```

Comment:
Here and in the other autofillers you can't add this check. If you do then it is no longer possible to have multiple "models" in a version directory. This is generally supported and especially required for TensorRT (see model_control.proto cc_model_filenames). I think you don't want any of these changes in the autofill.cc files.

Reply:
This check was already implied by (netdef_files.size() != 2) or (plan_files.size() != 1). So, this doesn't change the current behavior of the autofiller. However, as stated that we can have multiple "models" in the version directory, the solution would be to make autofiller parse the instance_group for gpu device(if none then select device id =0), get the compute capability, parse the cc_model_filenames and get the specific model filename to autofill against. Do we want to add this support in the autofiller?

Reply:
That means the autofiller wouldn't work with cc_model_filenames. I think the fix is to just change chose checks to make sure directory is not empty (or for netdef that there are at least 2 files). I would not get any fancier than that.